src/main/java/ladder/domain/LadderGame.java

```diff
@@ -0,0 +1,47 @@
+package ladder.domain;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import ladder.domain.ladder.Ladder;
+import ladder.domain.player.Player;
+import ladder.domain.player.Players;
+import ladder.domain.prize.Prize;
+import ladder.domain.prize.Prizes;
+
+public class LadderGame {
+    private final Map<Player, Prize> results;
+
+    private LadderGame(Map<Player, Prize> results) {
+        this.results = new LinkedHashMap<>(results);
+    }
+
+    public static LadderGame of(Players players, Ladder ladder, Prizes prizes) {
+        Map<Player, Prize> results = new LinkedHashMap<>();
+
+        for (int i = 0; i < players.size(); i++) {
+            Player player = players.getPlayers().get(i);
+            int endIndex = ladder.findEndIndex(i);
+            Prize prize = prizes.getPrizes().get(endIndex);
+
+            results.put(player, prize);
+        }
+
+        return new LadderGame(results);
+    }
+
+    public Prize getResultByPlayerName(String playerName) {
+        for (Entry<Player, Prize> entry : results.entrySet()) {
+            if (entry.getKey().name().equals(playerName)) {
+                return entry.getValue();
+            }
+        }
+
+        throw new IllegalArgumentException("존재하지 않는 참가자입니다.");
+    }
```

Comment:
혹시 시간 복잡도에 대해 알고 계신가요?
빅O 표기법으로 표현하자면 해당 코드는 O(1) 을 O(n)으로 풀어낸 코드예요 (Map 자료구조의 이점을 못 누리고 있음)

Player 키 값을 가지고 있으니 playerName을 Player 타입으로 바꾸어서 조회해주세용.
혹은 key를 String으로 교체하는 방법도 있겠죠. 컴퓨팅 성능이 좋아져서 작은 성능 이득 때문에 가독성을 포기할 필요는 없는 시대지만 있는 성능을 악화시키는 건 피해야합니다
```suggestion
    public Prize getResultByPlayerName(String playerName) {
        Player player = new Player(playerName);
        if (isEmpty(player)) {
            throw new IllegalArgumentException("존재하지 않는 참가자입니다.");
        }
        return results.get(player);
    }

    private boolean isEmpty(Player player) {
        return !results.containsKey(player);
    }
```

Reply:
앗... 실수네요. 

하나 궁금한 점이 있어요. 
!results.containsKey(player)이 충분히 이해가 간다고 생각해서 isEmpty로 분리를 하지 않았어요.
private이라도 메서드가 늘어나면 전체 코드가 읽기가 힘들기도 하고, 메서드로 분리할 경우 찾으러 가야하잖아요. 
웨지는 어떻게 생각하시나요?

Reply:
요건 부정연산자를 지양하자는 의미로 썼어요 ㅎ 부정연산자 단독으로 있으면 읽을만한데 저는 부정연산자랑 AND / OR 조건 하나만 섞이면 너무 어렵더라구요 ㅋㅋ
!isEmpty || valid(abc) 이런거요. 
혼자서 입으로 '빈게 아니면서.. 유효한건데.. 둘이 따로..' 하는데 부정연산자 섞여 있으면 30초씩 스턴 당합니다
요런 경험 몇번 하고나니 최대한 좀 !는 빼버리는 습관이 생겼어요

말씀하신 것처럼 `!results.containsKey(player)`가 메서드 분리할 필요없이 충분히 읽을만 하다는 데에는 동의합니다
분리해두면 굳이 cmd + b 찍어서 찾아서 체크해야하기도 하고요

근데 코드를 많이 봐야되는 상황이면 제 관심사가 아닌 부분들이 있거든요? 
예를들어 LadderGame 인원 중 두명이 중복 prize를 받는 버그가 났어요. 
그럼 버그를 찾기 위해 코드 흐름을 죽 눈으로 훑어야 되는데 `if( isEmpty(player) )` 이 부분의 조건까지는 안 궁금하니까 cmd + b 눌러보지도 않아요. 그냥 메서드명만 읽고 안에서 그런 검사를 할거라고 믿고 지나가는거죠 
이럴 땐 cmd + b도 안 찍어볼테니까, 오히려 잘 읽히는 표현이 메서드화 되어있는게 낫기도 하고요

그 작성해주신 코드 중에 조건문 좀 복잡한거 있었는데 index-1 뺀거랑 isExist()랑 || 비교한거.. 그런건 메서드로 빼는게 맞다는 확신이 있고요, 요런 상황에선 '!containsKey 하거나 메서드로 빼거나 둘다 아무래도 좋다' 라고 생각합니다 ㅎ